catalog/registry_test.go

```diff
@@ -88,6 +88,76 @@ func TestCatalogRegistry(t *testing.T) {
 	}, catalog.GetRegisteredCatalogs())
 }
 
+func TestRegistryPanic(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			assert.Equal(t, "catalog: RegisterCatalog catalog factory is nil", r)
+		} else {
+			t.Fatalf("expect a panic but did not get one")
+		}
+
+	}()
+	catalog.Register("foobar", nil)
+	t.Fatalf("expect a panic but Register function did not")
```

Comment:
`assert.PanicsWithValue(t, "catalog: RegisterCatalog catalog factory is nil", func() { catalog.Register("foobar", nil) })`

cleans this up and doesn't require you to manually do the defer etc.

Reply:
Thanks @zeroshade for the quick feedback, I have made the necessary changes :pray: